library/src/main/java/pl/netigen/netigenapi/BaseSplashActivity.kt

```diff
@@ -0,0 +1,138 @@
+package pl.netigen.netigenapi
+
+import android.os.Bundle
+import android.os.Handler
+
+import androidx.appcompat.app.AppCompatActivity
+
+import com.google.ads.consent.ConsentInfoUpdateListener
+import com.google.ads.consent.ConsentInformation
+import com.google.ads.consent.ConsentStatus
+import com.google.android.gms.ads.MobileAds
+
+import pl.netigen.rodo.ConstRodo
+import pl.netigen.rodo.GDPRDialogFragment
+
+abstract class BaseSplashActivity : AppCompatActivity(), ISplashActivity, AdmobIds, GDPRDialogFragment.GDPRClickListener {
+
+    var admobManager: AdmobManager? = null
+    var consentFinished: Boolean = false
+
+    private var gdprDialogFragment: GDPRDialogFragment? = null
+    private var canCommitFragment: Boolean = false
+    private lateinit var initAdmobHandler: Handler
+
+    abstract fun isNoAdsPaymentAvailable(): Boolean
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        contentView = contentView
+        Config.initialize(configBuilder)
+        if (!isNoAdsPaymentAvailable()) {
+            Config.setNoAdsBought(false)
+            showConsent()
+        }
+    }
+
+    private fun showConsent() {
+        val consentInformation = ConsentInformation.getInstance(this)
+        consentInformation.requestConsentInfoUpdate(publisherIds, object : ConsentInfoUpdateListener {
+            override fun onConsentInfoUpdated(consentStatus: ConsentStatus) {
+                val isInEea = ConsentInformation.getInstance(this@BaseSplashActivity).isRequestLocationInEeaOrUnknown
+                ConstRodo.setIsInEea(isInEea)
+                if (isInEea && consentInformation.consentStatus == ConsentStatus.UNKNOWN) {
+                    initRodoFragment()
+                } else {
+                    consentFinished = true
+                    startAdmobSplash()
+                }
+            }
+
+            override fun onFailedToUpdateConsentInfo(errorDescription: String) {
+                startAdmobSplash()
+            }
+        })
+    }
+
+    override fun onStop() {
+        super.onStop()
+        canCommitFragment = false
+        admobManager?.splashScreenOnStop()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        canCommitFragment = true
+        admobManager?.splashScreenOnStart()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        admobManager?.splashScreenOnDestroy()
+    }
+
+    private fun initRodoFragment() {
+        if (canCommitFragment) {
+            gdprDialogFragment = GDPRDialogFragment.newInstance()
+            gdprDialogFragment!!.setIsPayOptions(isNoAdsPaymentAvailable())
+            gdprDialogFragment!!.show(supportFragmentManager.beginTransaction().addToBackStack(null), "")
+        }
+    }
+
+    abstract fun showFullscreen(): Boolean
```

Comment:
This method should return a default value, could be moved to ISplashActivity interface.